File: CSharp/DP.Domain/Samples/Template/TrfFixing.cs
using System.Diagnostics;

namespace DP.Domain.Samples.Template
{
    public class TrfFixing : ProductFixingTemplate
    {
        protected override void FindWorkOptionLeg()
        {
            Trace.WriteLine("TRF: Find Working Option Leg");
        }
        protected override void CheckBarriers()
        {
            Trace.WriteLine("TRF: Check barries");
        }
        protected override bool RebateBarriers()
        {
            Trace.WriteLine("TRF: Rebate barries");
            return true;
        }

        protected override void FixingOptionLeg()
        {
            Trace.WriteLine("TRF: Fixing Option leg");
        }

        protected override void CheckTriggers()
        {
            Trace.WriteLine("TRF: Check Trigger!");
        }
    }
}s

File: Python/Samples/Template/TrfFixing.py
from ProductFixingTemplate import ProductFixingTemplate


class TrfFixing(ProductFixingTemplate):
    def findWorkOptionLeg(self):
        print("TRF: Find Working Option Leg!")

    def checkBarriers(self):
        print("TRF: Check barries!")

    def rebateBarriers(self):
        print("TRF: Rebate barries!")

    def fixingOptionLeg(self):
        print("TRF: Fixing Option leg!")

    def checkTriggers(self):
        print("TRF: Check Trigger!")
